### ISSUE TITLE
DataEditorRef.emit("copy") doesn't work when triggered from UI external to grid

### ISSUE THREAD

**Issue by raymeskhoury** [glideapps/glide-data-grid#446]
I'm trying to use DataEditorRef.emit("copy") to add a context menu to copy data from the grid. However, the data isn't getting copied. Debugging a bit it seems that the code in onCopy checks for [focus being within the grid's scrollable area or within the canvas](https://github.com/glideapps/glide-data-grid/blob/9030155da6bbc0f235617178997996678f860cd9/packages/core/src/data-editor/data-editor.tsx#L2514). However in the case of any button external to these elements, this is going to be false and the function early exits.

I might be misunderstanding/misusing the function though, let me know 😄

**Comment by jassmith** [comment]
ah yeah thats a good idea, we use it when the grid has internal focus so I just never hit this case. Will remove the check when emitting.

**Comment by jassmith** [comment]
fixed in 5.0

**Comment by city17** [comment]
It seems that the 'paste' emit event has the same issue?

I have the same use case as @raymeskhoury, and copy now works perfect (`dataEditorRef.current.emit('copy');`)

However, with paste, it doesn't work, but if I force focus on the grid right before the paste emit, it does work, like so:
```js
    dataEditorRef.current.focus();
    dataEditorRef.current.emit('paste');
```
